specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/examples/ExternalSubscriptionForecast.json

```diff
@@ -0,0 +1,82 @@
+{
+  "parameters": {
+    "api-version": "2019-11-01",
+    "externalCloudProviderId": "100",
+    "externalCloudProviderType": "externalSubscriptions",
+    "parameters": {
+      "type": "Usage",
+      "timeframe": "MonthToDate",
+      "dataset": {
+        "granularity": "Daily",
+        "filter": {
+          "and": [
+            {
+              "or": [
+                {
+                  "dimension": {
+                    "name": "ResourceLocation",
+                    "operator": "In",
+                    "values": [
+                      "East US",
+                      "West Europe"
+                    ]
+                  }
+                },
+                {
+                  "tag": {
+                    "name": "Environment",
+                    "operator": "In",
+                    "values": [
+                      "UAT",
+                      "Prod"
+                    ]
+                  }
+                }
+              ]
+            },
+            {
+              "dimension": {
+                "name": "ResourceGroup",
+                "operator": "In",
+                "values": [
+                  "API"
+                ]
+              }
+            }
+          ]
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "providers/Microsoft.CostManagement/externalSubscriptions/100/forecast/d99477af-7510-40ee-aca2-e59bdca0d10d",
```

Comment:
>providers/Microsoft.CostManagement/externalSubscriptions/100/forecast/d99477af-7510-40ee-aca2-e59bdca0d10d [](start = 15, length = 106)

What happens when the user does a GET on this?

Reply:
running a GET on this would return a 404

Reply:
@majastrz When a customer makes the POST call for the forecast there is no underlying resource, this is an extension verb of "Action" on the subscription to get the forecasted value for that subscription. The "id" in the response is merely a place holder in this case. What would you recommend us do ?

Reply:
On a POST call there's no requirement for the response to have the `id`, `name`, `type`, and `properties` properties. One option would be to not generate a fake ID.

One example of an RP that implements something similar is [https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json](https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/policyEvents.json). They don't return a fake resource in the response but instead just return a standard collection response.

---
In reply to: [419736378](https://github.com/Azure/azure-rest-api-specs/pull/9263#discussion_r419736378) [](ancestors = 419736378)

Reply:
And speaking of standard collection response, your nextLink is buried in an unexpected location.

Normally, a paging collection response would look like this:
```
{
  "value": [
    // objects making up one page
  ],
  "nextLink": "<optional continuation link"
}
```

Can we adjust your response to be more like the above?

---
In reply to: [419763695](https://github.com/Azure/azure-rest-api-specs/pull/9263#discussion_r419763695) [](ancestors = 419763695,419736378)

Reply:
This is the current contract which UX takes a dependency on. 
We'll be updating it in the next iteration.

Reply:
Makes sense.

---
In reply to: [419782152](https://github.com/Azure/azure-rest-api-specs/pull/9263#discussion_r419782152) [](ancestors = 419782152)